packages/api/src/task/scheduler.ts

```diff
@@ -99,12 +99,20 @@ export class TaskScheduler {
   }
 
   async processTaskEvent(event: messages.TaskResult): Promise<boolean> {
-    const tasks = await db.task.find({ id: event.task.id });
-    if (!tasks?.length || !tasks[0].length) {
-      console.log(`task event process error: task ${event.task.id} not found`);
-      return true;
+    let task = await db.task.get(event.task.id, { useReplica: false });
+    if (!task) {
+      console.log(
+        `task event process error: task ${event.task.id} not found, retrying after 500ms`,
+      );
+      await new Promise((resolve) => setTimeout(resolve, 500));
```

Comment:
I don't think this is necessary if we are already not using the replica. It is not possible to have inconsistencies when reading the primary, so if the task doesn't exist it won't be created later either. So this will overcomplicate the code unnecessarily IMO

Reply:
removed all the sleeps